### DISCUSSION TITLE
How to download weather_code for Ensemble Models?

### DISCUSSION THREAD

**Discussion by magheralex** [open-meteo/open-meteo#728] [Q&A] [answered]
Hi there,
How can I download the weather_code for GFS 0.5 & IFS 0.25 Ensemble Models?

`docker run -it --rm -v open-meteo-data:/app/data ghcr.io/open-meteo/open-meteo download-ecmwf --domain ifs025_ensemble --only-variables temperature_2m --run 0` downloads only the temperature_2m for ECMWF IFS025. The weather_code is not available as a variable.

Thanks in advance for your help.

**Comment by patrick-zippenfenig** [comment]
Except for DWD ICON models, weather codes are calculated on demand in the Open-Meteo API code. Unfortunately to calculate weather codes for IFS, pretty much all weather variables need to be downloaded. If you remove `--only-variables`, the weather code calculation will work in API calls

**Comment by magheralex** [comment]
So to use temperature_2m and weather_code for GFS 0.5 Ensemble, is it necessary to download all GFS 0.5 Ensemble variables? Do you know how much disk space is needed?

Another problem is that temperature_2m (GFS 0.5) alone takes about 1.5 hours to download (for ECMWF IFS 0.25 downloads in about 15 minutes.) I'm afraid I'll have to wait a day to download the entire data set.

Are you planning to upload datasets for Ensemble models to amazonaws in the near future?

**Reply by patrick-zippenfenig** [reply, marked as answer]
For GFS, a smaller list of weather variables is sufficient to get weather codes. See: https://github.com/open-meteo/open-meteo/blob/main/Sources/App/Gfs/GfsController.swift#L529

Currently there are no plans to also publish ensemble models on AWS. The amount of data is significantly larger. However, there might be derivatives like precipitation probabilities that could be added to AWS.

**Reply by magheralex** [reply]
Thanks Patrick.
